src/api/lib/tasks/dev/requests.rake

```diff
@@ -199,5 +199,357 @@ namespace :dev do
 
       puts "* Request with delete action #{request.number} has been created."
     end
+
+    desc 'Copy last 10 requests from openSUSE:Factory'
+    task requests_from_open_suse_factory: :development_environment do
+      require 'factory_bot'
+      include FactoryBot::Syntax::Methods
+
+      users = []
+      admin = User.get_default_admin
+      User.session = admin
+
+      # Setup interconnect
+      remote_proj = Project.find_by_name('openSUSE.org')
+      if remote_proj.nil?
+        print_message 'Creating the remote project for interconnect'
+
+        remote_proj = RemoteProject.new(name: 'openSUSE.org',
+                                        remoteurl: 'https://api.opensuse.org/public',
+                                        title: 'Standard OBS instance at build.opensuse.org',
+                                        description: 'This instance delivers the default build targets for OBS.')
+        remote_proj.store
+        FetchRemoteDistributionsJob.perform_now
+      end
+
+      clone_project(project_name: 'openSUSE:Factory', users: users)
+
+      # Get the list of requests from openSUSE:Factory
+      print_message 'Getting the list of requests'
+      url = "#{base_api_url}/request"
+      params = { view: 'collection', project: 'openSUSE:Factory', limit: '10', states: 'new,review' }
+      request = make_api_request(url: url, params: params)
+      requests_list = Xmlhash.parse(request)
+      print_message 'Successfully got the requests list'
+
+      requests_list['request'].each do |req|
+        # target project not openSUSE:Factory
+        next if req['action']['target']['project'] != 'openSUSE:Factory'
+        next if req['action']['target']['project'].match?('openSUSE:Factory:Staging')
+        next if req['action']['source']['project'].match?('openSUSE:Factory:Staging')
```

Comment:
How would any of this be possible?

Reply:
This requires exploring some more request data from the API. This would even make this PR hard to review and would even delay this PR. For now, we have good enough data to use in dev env. If we really need this then doing it in a follow-up pr would be the best way to move forward because staging requests can't use the logic we have in this PR for handling normal bs_requests.

Reply:
Ah that API returns also outgoing requests from `openSUSE:Factory` I see. That might mean you don't get *any* request that matches your criteria...

I guess you really want to use https://api.opensuse.org/apidocs/#/Search/get_search_request

Reply:
The API you mentioned also doesn't return what we are expecting. It just returns the "total number" of requests matched with criteria passed in parameters